scripts_src/maps/argarden.ssl

```diff
@@ -84,79 +82,67 @@ procedure map_update_p_proc begin
    Lighting;
 end
 
+//Extracted procedure to reduce duplication
+procedure Restock begin
+   variable item;
+
+   set_map_var(MVAR_Last_Respawn_Map,game_time);
+   call Make_Geckos;
+   call Make_Plants;
+   if (not(tile_contains_obj_pid(24298,0,PID_XANDER_ROOT))) then begin
+       item:=create_object_sid(PID_XANDER_ROOT,0,0,SCRIPT_AIXANDER);
+       move_to(item,24298,0);
+       set_map_var(MVAR_Current_Xander_Root,map_var(MVAR_Current_Xander_Root)+1);
+   end
+end
+
+procedure Spawn_Type(variable total, variable type, variable object_pid, variable object_script, variable mapvar) begin
+   variable counter:=0;
+   variable item;
+   variable item_hex:=0;
+
+   while (counter < total) do begin
```

Comment:
Better to use [for loops](https://sfall.bgforge.net/sslc/#additional-supported-syntax). While technically the same, they are more transparent.